packages/smooth_app/lib/helpers/text_input_formatters_helper.dart

```diff
@@ -10,14 +10,43 @@ import 'package:smooth_app/helpers/strings_helper.dart';
 /// on the user's language.
 ///
 /// It works if there are no group separator in the format, as it would be
-/// problematic with the `123,456.9` and both comma and dot.
+/// problematic with the `1,234,567.89` and both comma and dot.
 ///
 /// Also, if a separator is already displayed, it will be moved to the new
 /// position
 class DecimalSeparatorRewriter extends TextInputFormatter {
   DecimalSeparatorRewriter(NumberFormat format)
       : _decimalSeparator = format.symbols.DECIMAL_SEP,
-        _separatorToReplace = _findSeparatorToReplace(format);
+        _separatorToReplace = _findSeparatorToReplace(format) {
+    // Here we check that there are no group separators.
+    // The formatted string should
+    // * contain the nine digits 1..9
+    // * contain one decimal separator (either ',' or '.')
+    // * have a length of 10 (9 digits and one separator)
+    const double number = 1234567.89;
+    final String formatted = format.format(number);
+    if (formatted.length != 10) {
```

Comment:
So I would assume that we would want to tests several input strings, with a wide variety of inputs (e.g. "4,41", ".41", ".410", "0,41", "441" and even some string with groups: "4,430.21", "2 000", "2 000,32") etc.

Instead of checking the length, each digits etc. why not compare the resulting string to the expected string? e.g. something like  (formatted == "1234567.89") || (formatted == "1234567,89") ? That sounds much simpler, especially if we want to test all the different tests cases.

In fact maybe we can have a list of tests:

tests = (
([input string], [input decimal separator], [expected result string])

and then just iterate through the list of test and check if input string == expected string.

What do you think?

Reply:
The test here is _only_ about checking if there is a group separator: we don't really need a _list_ of tests.
Your suggestion `(formatted == '1234567.89') || (formatted == '1234567,89')` would work and is simpler to maintain.